docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,691 @@
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_io_index | integer | Index of the input or output that can be exitted.
+
+#### `invalid_piggyback`
+> A invalid_piggyback event
+
+```json
+{
+    "event": "invalid_piggyback",
+    "details": {
+        "type": "output",
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_io_index"  : 1,
+        "spending_blocknum"  : 10000,
+        "spending_tx_index"  : 10,
+        "spending_input_index"  : 0,
+    }
+}
+```
+
+An invalid piggyback is in process. Should be challenged.
+<aside class="warning"> Not Implemented Yet.</aside>
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+type | string | Indicates whether the invalid piggyback is on an "input" or "output"
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_io_index | integer | Index of the input or output of the in-flight transaction
+spending_blocknum | integer | Block number of the spending transaction
+spending_tx_index | integer | Index of the spending transaction
+spending_input_index | integer | Index of the spent input iofn the spending transaction
+
+
+## Inflight Exit - Get Exit Data
+
+```shell
+curl http://localhost:4000/inflight_exit.get_data -d '{"txhash": "bdf562c24ace032176e27621073df58ce1c6f65de3b5932343b70ba03c72132d"}'
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "txbytes": "F847010180808080940000...",
+        "sigs": "7C29FB8327F60BBFC62...",
+        "input_txs" : [
+            {
+                "txbytes": "F81891018080808...",
+                "proof": "CEDB8B31D1E4C..."
+            },
+            {
+                "txbytes": "2A03418086001...",
+                "proof": "A67131D1E4C..."
+            }
+        ]
+    }
+}
+```
+
+Gets exit data for an in-flight exit
+
+### HTTP Request
+
+`POST /inflight_exit.get_data`
+
+### Request Body
+
+Attribute | Type | Description
+--------- | ------- | -----------
+txhash | Hex encoded string | The hash of the in-flight transaction
+
+
+
+## Inflight Exit - Get Challenge Data
+
+```shell
+curl http://localhost:4000/inflight_exit.get_challenge_data -d '{"txhash": "bdf562c24ace032176e27621073df58ce1c6f65de3b5932343b70ba03c72132d"}'
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "inflight_txbytes": "F847010180808080940000...",
+        "inflight_input_index": 1,
+        "competing_txbytes": "F317010180808080940000...",
+        "competing_input_index": 1,
+        "competing_txid": 2600003920012,
```

Comment:
also note that `competing_input_index` might be 0, which means that no proof is there and the challenge uses a non-included transaction (or a transaction which inclusion is unknown)